File: samples/CS/19-Localization/Program.cs
//#define german
//comment or uncomment line before to change language
using System;
using System.Collections.Generic;
using System.Linq;
using System.Windows.Forms;
using System.Threading;
using System.Globalization;

namespace _19_Localization
{
    static class Program
    {
        /// <summary>
        /// The main entry point for the application.
        /// </summary>
        [STAThread]
        static void Main()
        {
#if german
            Thread.CurrentThread.CurrentCulture = new CultureInfo("de-DE");
            Thread.CurrentThread.CurrentUICulture = new CultureInfo("de-DE");
#else
            Thread.CurrentThread.CurrentCulture = new CultureInfo("en-us");
            Thread.CurrentThread.CurrentUICulture = new CultureInfo("en-us");
#endif
#if NET462_OR_GREATER
            Application.EnableVisualStyles();
            Application.SetCompatibleTextRenderingDefault(false);
#else
            // To customize application configuration such as set high DPI settings or default font,
            // see https://aka.ms/applicationconfiguration.
            ApplicationConfiguration.Initialize();
#endif
#if NET10_0_OR_GREATER
            Application.SetColorMode(SystemColorMode.System);
#endif
            Application.Run(new Form1());
        }
    }
}

File: samples/VB/19-Localization/Program.vb
'#Const german = 1
'comment or uncomment line before to change language
Imports System.Threading
Imports System.Globalization

Namespace _19_Localization
	Friend NotInheritable Class Program
		''' <summary>
		''' The main entry point for the application.
		''' </summary>
		Private Sub New()
		End Sub
		<STAThread>
		Shared Sub Main()
#If german Then
			Thread.CurrentThread.CurrentCulture = New CultureInfo("de-DE")
			Thread.CurrentThread.CurrentUICulture = New CultureInfo("de-DE")
#Else
			Thread.CurrentThread.CurrentCulture = New CultureInfo("en-us")
			Thread.CurrentThread.CurrentUICulture = New CultureInfo("en-us")
#End If
			Application.EnableVisualStyles()
			Application.SetCompatibleTextRenderingDefault(False)
			Application.Run(New Form1())
		End Sub
	End Class
End Namespace
